src/posix/platform/radio.cpp

```diff
@@ -88,7 +88,6 @@ Radio::Radio(const char *aUrl)
 void Radio::Init(void)
 {
     bool        resetRadio             = (mRadioUrl.GetValue("no-reset") == nullptr);
-    bool        restoreDataset         = (mRadioUrl.GetValue("ncp-dataset") != nullptr);
```

Comment:
Maybe we can add a check that if URL happens to contain the `"ncp-dataset"` we print an error (and/or exit) indicating that this feature is removed?

Reply:
Updated